core/src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -42,8 +44,8 @@ trait CassandraRecovery extends CassandraTagRecovery with TaggedPreparedStatemen
           for {
             tp <- lookupTagProgress(persistenceId)
             _ <- sendTagProgress(persistenceId, tp, tagWrites.get)
-            startingSequenceNr = calculateStartingSequenceNr(tp)
-            _ <- sendPreSnapshotTagWrites(startingSequenceNr, fromSequenceNr, persistenceId, Long.MaxValue, tp)
+            scanningSeqNr <- tagScanningStartingSequenceNr(persistenceId, tp)
```

Comment:
I think this could happen in concurrently with the sendTagProgress but don't think it will speed much up

Reply:
good suggestion, fixed